src/main/java/org/assertj/core/api/AbstractIterableAssert.java

```diff
@@ -143,6 +146,113 @@ public SELF isNotEmpty() {
+   * it also enforces type compatibility of the elements contained in those Iterables. For an example, see {@link RecursiveIterableComparisonAssert#isEqualTo(Iterable)}.</li>
+   * <li>
+   * although different type of iterables can be compared, order does matter. Comparing a List<Person> to a Set<Person> will almost always fail,
+   * but the order of comparison can be configured to be ignored by calling {@link RecursiveIterableComparisonAssert#ignoringCollectionOrder()} () ignoringCollectionOrder}.
+   * This allows the comparison of ordered/unordered iterable pairs, such as comparing a a List<Person> to a Set<Person>.
+   * Note that calling {@link RecursiveIterableComparisonAssert#ignoringCollectionOrder} also causes the order of any fields/properties that are iterables/collections to be ignored.
+   * </li>
+   * </ul>
+   * </p>
+   *
+   * <p>
+   * It is also possible to change the behavior of comparison just as the way {@link RecursiveComparisonAssert RecursiveComparisonAssert} allows:
+   * <ul>
+   *   <li>Choosing a strict or lenient recursive comparison (lenient being the default which allows to compare different types like {@code Doctor} and {@code Person} </li>
+   *   <li>Ignoring fields in the comparison </li>
+   *   <li>Specifying comparators to use in the comparison per fields and types</li>
+   *   <li>Forcing recursive comparison on classes that have redefined equals (by default overridden equals are used)</li>
+   * </ul>
+   * <p>
+   * Please consult the detailed documentation available here: <a href="https://assertj.github.io/doc/#assertj-core-recursive-comparison">https://assertj.github.io/doc/#assertj-core-recursive-comparison</a>
+   *
+   * @return a new {@link RecursiveIterableComparisonAssert} instance
+   */
+  @Beta
+  public RecursiveIterableComparisonAssert<?, ACTUAL> usingRecursiveComparison() {
+    return usingRecursiveComparison(new RecursiveComparisonConfiguration());
+  }
+
+  /**
+   * Same as {@link #usingRecursiveComparison()} but allows to specify your own {@link RecursiveComparisonConfiguration}.
+   * @param recursiveComparisonConfiguration the {@link RecursiveComparisonConfiguration} used in the chained {@link RecursiveIterableComparisonAssert} assertions.
+   *
+   * @return a new {@link RecursiveIterableComparisonAssert} instance built with the given {@link RecursiveComparisonConfiguration}.
+   */
+  @Beta
+  public RecursiveIterableComparisonAssert<?, ACTUAL> usingRecursiveComparison(RecursiveComparisonConfiguration recursiveComparisonConfiguration) {
+    SELF myselfCopy = usingElementComparator(new RecursiveComparisonDifferenceComparator(recursiveComparisonConfiguration));
+    return new RecursiveIterableComparisonAssert<>(actual, recursiveComparisonConfiguration, myselfCopy)
+      .withAssertionState(myselfCopy)
+      .withTypeComparators(comparatorsByType);
```

Comment:
remove the call to `withTypeComparators` as the `RecursiveComparisonConfiguration` provides the type comparators (and much more).